nuxeo-features/nuxeo-admin-center/nuxeo-admin-center-core/src/main/java/org/nuxeo/ecm/admin/operation/HotReloadStudioSnapshot.java

```diff
@@ -71,46 +80,110 @@
     protected boolean validate = true;
 
     @OperationMethod
-    public void run() throws Exception {
+    public Blob run() {
+
+        JSONArray result = new JSONArray();
+        JSONObject resultJSON = new JSONObject();
+
         if (updateInProgress) {
-            return;
+            resultJSON.put("status", "inProgress");
+            result.add(resultJSON);
+            return Blobs.createJSONBlob(result.toString());
         }
 
         if (!((NuxeoPrincipal) session.getPrincipal()).isAdministrator()) {
-            throw new NuxeoException("Must be Administrator to use this function");
+            resultJSON.put("status", "error");
+            resultJSON.put("message", "Must be Administrator to use this function");
```

Comment:
Create a method which wraps these two lignes and the instantiation of the `JSON` objects